macros/vc.asm

```diff
@@ -0,0 +1,83 @@
+vc_hook: MACRO
+if DEF(_RED_VC) || DEF(_BLUE_VC)
+.VC_\1::
+endc
+ENDM
+
+vc_patch: MACRO
+if DEF(_RED_VC) || DEF(_BLUE_VC)
+	assert !DEF(CURRENT_VC_PATCH), "Already started a vc_patch"
+CURRENT_VC_PATCH EQUS "\1"
+.VC_{CURRENT_VC_PATCH}::
+endc
+ENDM
+
+vc_patch_end: MACRO
+if DEF(_RED_VC) || DEF(_BLUE_VC)
+	assert DEF(CURRENT_VC_PATCH), "No vc_patch started"
+.VC_{CURRENT_VC_PATCH}_End::
+	PURGE CURRENT_VC_PATCH
+endc
+ENDM
+
+vc_assert: MACRO
+if DEF(_RED_VC) || DEF(_BLUE_VC)
+	assert \#
+endc
+ENDM
+
+vc_red_hook: MACRO
+if DEF(_RED_VC)
+.VC_\1::
+endc
+ENDM
+
+vc_red_patch: MACRO
+if DEF(_RED_VC)
```

Comment:
Hopefully these red and blue macros can be eliminated, but if not, they should still avoid repeating the same internals:

```
vc_red_patch: MACRO
IF DEF(_RED_VC)
	vc_patch \1
ENDC
ENDM
```